statics/js/components/topology.js

```diff
@@ -271,6 +294,28 @@ var TopologyComponent = {
       this.layout.zoomFit();
     },
 
+    topologyFilterClear: function () {
+         $("#topology-filter").attr('placeholder',$("#topology-filter").val());
+         $("#topology-filter").val('');
+     },
+
+    topologyFilterMouseLeave: function () {
+       if ($("#topology-filter").val() == '') {
+         $("#topology-filter").val($("#topology-filter").attr('placeholder'));
+       }
+    },
+
+    endsWith: function (str, suffix) {
+       return str.indexOf(suffix, str.length - suffix.length) !== -1;
+    },
+
+    topologyFilterQuery: function() {
+        if ($("#topology-filter").val() == '' || this.endsWith($("#topology-filter").val(), "SubGraph()")) {
+            this.$store.commit('topologyFilter', this.topologyFilter);
+            this.syncTopo();
```

Comment:
You probably want to debounce this method to avoid syncing the topo on each keypress

Reply:
I think syncTopo is debounced... (line 143 in topology.js)
I am very fresh to js.. so maybe I dont understand it well, sorry...